src/vcfio/utils/str_utils.py

```diff
@@ -33,12 +31,46 @@ def standardize_chromosome(raw_chr):
     return chromosome
 
 
-def calculate_zygosity(gt: AnyStr) -> Zygosity:
+def calculate_zygosity(genotype: str) -> Optional[Zygosity]:
     """
     calculate zygosity of a sample
     examples:
         '0/1' -> HET
         './.' -> NO_COVERAGE
     """
-    gt_pair = GT_PATTERN.findall(gt)[0]
-    return GT_ZYGOSITY_MAP.get(gt_pair, Zygosity.no_coverage)
\ No newline at end of file
+    zygosity = None
+    if len(genotype) == 1:
+        zygosity = _parse_haploid(genotype)
+    elif len(genotype) == 3:
+        zygosity = _parse_diploid(genotype)
+
+    return zygosity
+
+
+def _parse_haploid(genotype: str) -> Optional[Zygosity]:
+    if genotype == '0':
+        return Zygosity.reference
+    elif genotype.isdigit():
+        return Zygosity.homozygote
+    else:
+        return None
```

Comment:
Not a change request but just in other places I suggest raising an exception if we see an unexpected value. Here we can do something like
```
if genotype == '.':
    return None
raise IncorrectGenotypeException
```